docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1152 @@
+========== ==================================================
+101        '{"comment" : "nice", "kids" : [10, 13, 16]}'
+102        '{"comment" : "problematic", "kids" : [8, 11]}'
+103        '{"comment" : "knows best", "kids" : [2]}'
+========== ==================================================
+
+The following query gets the ``kids`` array for each customer:
+
+.. code-block:: text
+
+    SELECT id, json_query(
+                          description,
+                          'lax $.kids'
+                         ) kids
+    FROM customers
+
+========== ================
+id         kids
+========== ================
+101        '[10,13,16]'
+102        '[8,11]'
+103        '[2]'
+========== ================
+
+The following query gets the collection of kids for each customer. Note that
+the ``json_query`` function can only output a single JSON item. If we don't use
+array wrapper, we get error for every customer with multiple kids. The error is
+handled according to the ``ON ERROR`` clause.
+
+.. code-block:: text
+
+    SELECT id, json_query(
+                          description,
+                          'lax $.kids[*]'
+                          WITHOUT ARRAY WRAPPER
+                          NULL ON ERROR
+                         ) kids
+    FROM customers
+
+========== ================
+id         kids
+========== ================
+101        NULL
+102        NULL
+103        '2'
+========== ================
+
+The following query gets the last kid for each customer, wrapped in a JSON
+array:
+
+.. code-block:: text
+
+    SELECT id, json_query(
+                          description,
+                          'lax $.kids[last]'
+                          WITH ARRAY WRAPPER
+                         ) last_kid
+    FROM customers
+
+========== ================
+id         last_kid
+========== ================
+101        '[16]'
+102        '[11]'
+103        '[2]'
+========== ================
+
+The following query gets all kids above the age of 12 for each customer,
+wrapped in a JSON array. The second and the third customer don't have kids of
+this age. Such case is handled according to the ``ON EMPTY`` clause. Default
+``ON EMPTY`` returned value is ``NULL``. In the following example, ``EMPTY
+ARRAY ON EMPTY`` is specified.
+
+.. code-block:: text
+
+    SELECT id, json_query(
+                          description,
+                          'strict $.kids[*]?(@ > 12)'
+                          WITH ARRAY WRAPPER
+                          EMPTY ARRAY ON EMPTY
+                         ) kids
+    FROM customers
+
+========== ================
+id         kids
+========== ================
+101        '[13,16]'
+102        '[]'
+103        '[]'
+========== ================
+
+The following query shows the result of the ``QUOTES`` clause. Note that ``KEEP
+QUOTES`` is the default.
+
+.. code-block:: text
+
+    SELECT
+          id,
+          json_query(description, 'strict $.comment' KEEP QUOTES) quoted_comment,
```

Comment:
AS quotes_comment,

Reply:
I don't use `AS` for aliases in this document.

Reply:
I know .. but we should consistently use it. Its clearer for the reader.